repair/repair.cc

```diff
@@ -622,45 +622,49 @@ repair_neighbors repair::shard_repair_task_impl::get_repair_neighbors(const dht:
         neighbors[range];
 }
 
-size_t repair::shard_repair_task_impl::ranges_size() {
+size_t repair::shard_repair_task_impl::ranges_size() const noexcept {
     return ranges.size() * table_ids.size();
 }
 
 // Repair a single local range, multiple column families.
 // Comparable to RepairSession in Origin
-future<> repair::shard_repair_task_impl::repair_range(const dht::token_range& range, ::table_id table_id) {
-    check_in_abort_or_shutdown();
-    ranges_index++;
-    repair_neighbors r_neighbors = get_repair_neighbors(range);
+future<> repair::table_repair_task_impl::repair_range(const dht::token_range& range) {
+    _local_repair_data.check_in_abort_or_shutdown();
+    _local_repair_data.ranges_index++;
```

Comment:
We already have the range repair task, yet we still need to track ranges inside the shard repair task.  Can this be passed to range repair task from shard task directly?

Reply:
ranges_index counts the ranges for whole repair on one shard and not for specific table so it can't be directly moved.